indexer/ai/src/main/java/com/google/cloudsearch/LocalRepository.java

```diff
@@ -81,19 +85,14 @@ public void init(RepositoryContext repositoryContext) throws RepositoryException
 
     @Override
     public CheckpointCloseableIterable<ApiOperation> getAllDocs(@Nullable byte[] bytes) throws RepositoryException {
-        //Iterate through all resources and execute skills for each resource
-        /*
-            Pseudo Code:
 
-            for each resource:
-                for each skill in aiSkillSet:
-                    executeSkill(resourcePath)
-         */
+
+        skillDriver = new AISkillDriver(aiConfig, schema);
```

Comment:
why are we initializing driver within local repository ? I think what we discussed is that driver will take the repository and all, and will run the pipeline.

Reply:
I interpreted it as driver does all the work related to managing the AISkills, I didn't realize you meant to have like a top level thing,  but I feel even this is easy to use and is just a function call to fill out the structured data. I also think it fits in the pipeline, as the structured data are supposed to be filled when each document is indexed, and therefore for each document the connector needs to call this function.